process/block/preprocess/sortedTransactionsProvider.go

```diff
@@ -0,0 +1,65 @@
+package preprocess
```

Comment:
As stated at 8), this PR removes the old, deprecated sorting operations of the old transactions cache.

`preprocess/transactions` handles the `TxCache` using the `Cacher` interface. However, when it selects transactions at proposal time, it requires the **selection** operation, which the `Cacher` interface does not provide. This is why we needed the adapter in the first place. The adapter factory was switching between the new `sortedTxsProvider` and the old one - the old one was used if the cast `cache.(*TxCache)` failed (never the case in real life). Now, since we've removed the old components, we had to use a disabled substitute. Of course, this is suboptimal design. We actually have to fix `transactions.go` to depend on a **well-defined** interface instead, and discard the adapters and the adapter factory altogether. Not doable in this PR though, I've added a TODO now. I will try to figure out another way in the next PR, `pool++cross`.

Furthermore, the factory and the adapters were kept in separate files before this PR. I grouped them together to increase the cohesion of _smelly things that we should remove sometimes soon_. 

Is this reasonable (that is, doing this non-trivial removal / change incrementally to avoid larger changes in the code at once)?

Reply:
It is reasonable as it is for now.